### PR TITLE
Fix: Stuff Management

### PR DESCRIPTION
**Pull request by sonylomo**
## Description

- Implement edit stuff feature and populate the data to be edited on the form
- Removed redundant column filters 
- Form value reset on form submission and query invalidation for _all stuff details_ 

## Motivation and Context

- Fixes issue #[KAN-27](https://pamba-africa.atlassian.net/browse/KAN-27?atlOrigin=eyJpIjoiNGFhMWJmNGJlNTY4NDhmMTg4OTZkNmQzMmJmMWY0OTAiLCJwIjoiaiJ9)

## How Has This Been Tested?

- Tested edit, delete and creation actions locally

## Screenshots (if appropriate):

- N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,6 +27,7 @@ yarn-error.log*
 
 # local env files
 .env*.local
+.env
```
Comment (daveclinton) [inline .gitignore:30]: we are using the env.local
why add the extra file?

File: app/api/requests.ts [hunk 2]
```diff
@@ -174,39 +144,100 @@ export const useDeleteExpense = () => {
   });
 };
 
-export const useDeleteStaff = () => {
+export const useEditExpense = () => {
   const dispatch = useAppDispatch();
-  return useMutation<void, Error, any>(async (staff_id: number) => {
+  return useMutation<void, Error, any>(async ([expenxeId, formData]) => {
```
Comment (daveclinton) [inline app/api/requests.ts:149]: add a appropriate response type

File: app/api/requests.ts [hunk 3]
```diff
@@ -174,39 +144,100 @@ export const useDeleteExpense = () => {
...
+    } catch (error) {
+      throw new Error("Unable to fetch Staff");
+    }
+  });
+};
+
+export const useCreateStaff = () => {
   const dispatch = useAppDispatch();
-  return useMutation<void, Error, any>(async ([expenxeId, formData]) => {
+  return useMutation<void, Error, any>(async ({ f_name, phone, role }) => {
```
Comment (daveclinton) [inline app/api/requests.ts:183]: add appropriate payload type

File: app/api/requests.ts [hunk 4]
```diff
@@ -174,39 +144,100 @@ export const useDeleteExpense = () => {
...
-      dispatch(setMessage(""));
-    }, 3000);
     return response;
   });
 };
 
+export const useDeleteStaff = () => {
+  const queryClient = useQueryClient();
+  const dispatch = useAppDispatch();
+  return useMutation<void, Error, any>(
```
Comment (daveclinton) [inline app/api/requests.ts:198]: add appropriate response type

File: app/api/requests.ts [hunk 5]
```diff
@@ -174,39 +144,100 @@ export const useDeleteExpense = () => {
...
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries("getAllStaff");
+      },
+    }
+  );
+};
+
+export const useEditStaff = () => {
+  const queryClient = useQueryClient();
```
Comment (daveclinton) [inline app/api/requests.ts:219]: do we need to cache a query here? Or what is the query client for here?

**Review by daveclinton — CHANGES_REQUESTED** [review]

**Comment by daveclinton** [comment]
You should also create your feature branch off release branch @sonylomo

**Review by McbullIan — APPROVED** [review]
LGTM

**Review by regan-mu — DISMISSED** [review]

**Review by daveclinton — APPROVED** [review]

**Pull request merged**